builder/dockerignore.go

```diff
@@ -26,8 +26,8 @@ type DockerIgnoreContext struct {
 //
 // TODO: Don't require a ModifiableContext (use Context instead) and don't remove
 // files, instead handle a list of files to be excluded from the context.
-func (c DockerIgnoreContext) Process(filesToRemove []string) error {
-	f, err := c.Open(".dockerignore")
+func (c DockerIgnoreContext) Process(fileName string, filesToRemove []string) error {
```

Comment:
Can we call this `ignoreFile`, its very confusing to have to work out what `fileName` refers to.